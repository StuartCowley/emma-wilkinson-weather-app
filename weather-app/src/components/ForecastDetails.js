import React from "react";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__temperature">
        <span className="forecast-details__label"> Max Temperature: </span>
        <span className="forecast-details__temperature-max">
          {temperature.max}&deg;C{"  "}
        </span>

        <span className="forecast-details__label"> Min Temperature: </span>
        <span className="forecast-details__temperature-min">
          {temperature.min}&deg;C
        </span>
      </div>
      <div className="forecast-details_humidity">
        <span className="forecast-details__label"> Humidity: </span>
        <span className="forecast-details__humidity"> {humidity} % </span>
      </div>
      <div className="forecast-details__wind">
        <span className="forecast-details__label"> Wind: </span>
        <span className="forecast-details__wind-direction">
          {wind.direction}
        </span>
        <span className="forecast-details__wind-speed"> {wind.speed} mph </span>
      </div>
    </div>
  );
}

export default ForecastDetails;
