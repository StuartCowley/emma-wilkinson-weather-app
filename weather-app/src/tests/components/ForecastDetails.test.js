import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 18,
      direction: "n",
    },
    humidity: 20,
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("render correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("12°C")).toHaveClass("forecast-details__temperature-min");
    expect(getByText("22°C")).toHaveClass("forecast-details__temperature-max");
    expect(getByText("18 mph")).toHaveClass("forecast-details__wind-speed");
    expect(getByText("n")).toHaveClass("forecast-details__wind-direction");
    expect(getByText("20 %")).toHaveClass("forecast-details__humidity");
  });
});
