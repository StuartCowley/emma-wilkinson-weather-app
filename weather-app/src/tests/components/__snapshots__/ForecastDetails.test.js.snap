// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature"
    >
      <span
        class="forecast-details__label"
      >
         Max Temperature: 
      </span>
      <span
        class="forecast-details__temperature-max"
      >
        22°C  
      </span>
      <span
        class="forecast-details__label"
      >
         Min Temperature: 
      </span>
      <span
        class="forecast-details__temperature-min"
      >
        12°C
      </span>
    </div>
    <div
      class="forecast-details_humidity"
    >
      <span
        class="forecast-details__label"
      >
         Humidity: 
      </span>
      <span
        class="forecast-details__humidity"
      >
         20 % 
      </span>
    </div>
    <div
      class="forecast-details__wind"
    >
      <span
        class="forecast-details__label"
      >
         Wind: 
      </span>
      <span
        class="forecast-details__wind-direction"
      >
        n
      </span>
      <span
        class="forecast-details__wind-speed"
      >
         18 mph 
      </span>
    </div>
  </div>
</DocumentFragment>
`;
